diff -ru lesstif-0.95.2.orig/scripts/Makefile.in lesstif-0.95.2/scripts/Makefile.in
--- lesstif-0.95.2.orig/scripts/Makefile.in	2010-03-25 10:52:48.507323101 +0100
+++ lesstif-0.95.2/scripts/Makefile.in	2010-03-25 10:54:36.234134684 +0100
@@ -287,7 +287,7 @@
 xmVERSION = @xmVERSION@
 xmVer = @xmVer@
 MAINTAINERCLEANFILES = Makefile.in typedefs.c
-SUBDIRS = . FreeBSD RedHat Slackware autoconf OS2
+SUBDIRS = . FreeBSD RedHat Slackware OS2
 EXTRA_DIST = LTsysinfo
 dist_bin_SCRIPTS = motif-config
 typedefs_SOURCES = typedefs.l
