diff -ru lesstif-0.95.2.orig/configure lesstif-0.95.2/configure
--- lesstif-0.95.2.orig/configure	2012-08-20 11:56:42 +0200
+++ lesstif-0.95.2/configure	2012-08-20 11:57:35 +0200
@@ -19826,7 +19826,7 @@
 
 if test -x "$man2html_cmd"
 then
-  if test -z "`$man2html_cmd </dev/null 2>&1 | grep \"LessTif's man2html\"`"
+  if test -z "`$man2html_cmd </dev/null 2>&1 | grep \"LessTif\'s man2html\"`"
   then
           MAN2HTML="\$(top_builddir)/scripts/man2html"
      { $as_echo "$as_me:$LINENO: result: Use LessTif one" >&5
