--- libconfix/plugins/automake/configure.py.orig	2013-01-11 14:11:07 +0100
+++ libconfix/plugins/automake/configure.py	2013-01-11 14:16:43 +0100
@@ -27,7 +27,12 @@
     assert type(prefix) in [types.NoneType, types.ListType, types.TupleType]
     assert type(readonly_prefixes) in [types.NoneType, types.ListType, types.TupleType]
     
+    program = os.sep.join(packageroot + ['configure'])
     argv = []
+    if 'CONFIG_SHELL' in os.environ:
+        argv.append(program)
+        program = os.environ['CONFIG_SHELL']
+        pass
     if prefix is not None:
         argv.append('--prefix='+os.sep.join(prefix))
         pass
@@ -45,7 +50,7 @@
         argv.extend(args)
         pass
 
-    external_cmd.exec_program(program=os.sep.join(packageroot + ['configure']),
+    external_cmd.exec_program(program=program,
                               args=argv,
                               env=env,
                               dir=builddir,
