? .cproject
Index: src/ClientHandler.cc
===================================================================
RCS file: /cvsrootb/saloon/wamas_x/utilities/logging/cpp/logtool/src/ClientHandler.cc,v
retrieving revision 1.21
diff -u -r1.21 ClientHandler.cc
--- src/ClientHandler.cc	14 Aug 2013 11:38:57 -0000	1.21
+++ src/ClientHandler.cc	10 Sep 2013 14:02:43 -0000
@@ -237,6 +237,9 @@
 #endif
 
     std::string errorFile=getenv(LT_ENV_ROOTDIR);
+    if(!errorFile.empty()) {
+    	errorFile+="/";
+    }
     errorFile+=LOG_SERVERERRORS;
     errFile.open(errorFile.c_str(), std::ios::out|std::ios::app);
         
