? .cproject
Index: src/LogConnection.cc
===================================================================
RCS file: /cvsrootb/saloon/wamas_x/utilities/logging/cpp/logtool/src/LogConnection.cc,v
retrieving revision 1.29
diff -u -r1.29 LogConnection.cc
--- src/LogConnection.cc	12 Dec 2013 16:02:45 -0000	1.29
+++ src/LogConnection.cc	9 Jan 2014 08:50:52 -0000
@@ -114,9 +114,10 @@
 
 bool LogConnection::open()
 {
-    if ( sock_ !=  INVALID_SOCKET )
-        return connected_;
-    
+	if ( sock_ !=  INVALID_SOCKET) {
+		close();
+	}
+
     if ( local_.empty() ) {
         char            localHost[MAXHOSTNAMELEN+1];
         gethostname(localHost, MAXHOSTNAMELEN);
