? .cproject
Index: src/LogClientLogger.cc
===================================================================
RCS file: /cvsrootb/saloon/wamas_x/utilities/logging/cpp/logtool/src/LogClientLogger.cc,v
retrieving revision 1.18
diff -u -r1.18 LogClientLogger.cc
--- src/LogClientLogger.cc	14 May 2012 13:17:52 -0000	1.18
+++ src/LogClientLogger.cc	19 Jul 2013 06:26:16 -0000
@@ -71,6 +71,7 @@
         if ( ! conn_.connected() ) {
             LogConfPtr conf(new LocalConf());
             LogConfMgr::instance().conf(conf);
+            logMsg->message("<LOCAL_LOG> "+logMsg->message()+" <LOCAL_LOG>");
             LogWriter::instance().write(logMsg);
         } 
     } 
