--- logtool-2.2.2/src/LogDriver.cc.orig	2011-04-29 10:26:00.419897215 +0200
+++ logtool-2.2.2/src/LogDriver.cc	2011-04-29 10:30:42.190533944 +0200
@@ -34,6 +34,22 @@
 
     len = message.size(); 
     
+    /* replace '\n' */
+	pos = 0;
+    while ( (pos = message.find("\\n", pos)) <= len ) {
+        message.replace(pos, 2, 1, '\n');
+        len = message.size();
+		pos += 1;
+    }
+    
+    /* replace '\t' */
+	pos = 0;
+    while ( (pos = message.find("\\t", pos)) <= len ) {
+        message.replace(pos, 2, 1, '\t');
+        len = message.size();
+		pos += 1;
+    }
+
     /* replace date */
 	pos = 0;
     while ( (pos = message.find("%D", pos)) <= len ) {
@@ -110,21 +126,6 @@
 		pos += msg->message().size();
     }
     
-    /* replace '\n' */
-	pos = 0;
-    while ( (pos = message.find("\\n", pos)) <= len ) {
-        message.replace(pos, 2, 1, '\n');
-        len = message.size();
-		pos += 1;
-    }
-    
-    /* replace '\t' */
-	pos = 0;
-    while ( (pos = message.find("\\t", pos)) <= len ) {
-        message.replace(pos, 2, 1, '\t');
-        len = message.size();
-		pos += 1;
-    }
     fputs(message.c_str(), fp);
 }
 
