--- logtool-2.2.2/src/ClientHandler.cc.orig	2010-12-17 14:38:33.468154121 +0100
+++ logtool-2.2.2/src/ClientHandler.cc	2010-12-17 14:42:51.081397724 +0100
@@ -2,6 +2,10 @@
  * @file
  * @author Copyright (c) 2006 Salomon Automation GmbH
  */
+#ifdef _WIN32
+/* bug#24499 comment#4 */
+#define FD_SETSIZE 1024
+#endif
 
 #include "ClientHandler.h"
 
--- logtool-2.2.2/src/LogClientLogger.cc.orig	2010-12-17 14:38:37.704678031 +0100
+++ logtool-2.2.2/src/LogClientLogger.cc	2010-12-17 14:42:55.844716233 +0100
@@ -2,6 +2,10 @@
  * @file
  * @author Copyright (c) 2006 Salomon Automation GmbH
  */
+#ifdef _WIN32
+/* bug#24499 comment#4 */
+#define FD_SETSIZE 1024
+#endif
  
 #include "LogClientLogger.h"
 
--- logtool-2.2.2/src/LogConnection.cc.orig	2010-12-17 14:38:43.234693793 +0100
+++ logtool-2.2.2/src/LogConnection.cc	2010-12-17 14:43:05.037983877 +0100
@@ -2,6 +2,10 @@
  * @file
  * @author Copyright (c) 2006 Salomon Automation GmbH
  */
+#ifdef _WIN32
+/* bug#24499 comment#4 */
+#define FD_SETSIZE 1024
+#endif
 
 #include "LogConnection.h"
 
--- logtool-2.2.2/src/LogLine.cc.orig	2010-12-17 14:38:46.631420854 +0100
+++ logtool-2.2.2/src/LogLine.cc	2010-12-17 14:43:09.401329971 +0100
@@ -2,6 +2,10 @@
  * @file
  * @author Copyright (c) 2006 Salomon Automation GmbH
  */
+#ifdef _WIN32
+/* bug#24499 comment#4 */
+#define FD_SETSIZE 1024
+#endif
 
 #include "LogLine.h"
 
--- logtool-2.2.2/src/LogServer.cc.orig	2010-12-17 14:41:55.074976246 +0100
+++ logtool-2.2.2/src/LogServer.cc	2010-12-17 14:43:47.604659760 +0100
@@ -3,6 +3,14 @@
  * @author Copyright (c) 2006 Salomon Automation GmbH
  */
 
+/* bug#24499 comment#4 */
+#ifndef _WIN32
+#include <sys/time.h>
+#include <sys/resource.h>
+#else
+#define FD_SETSIZE 1024
+#endif
+
 #include "LogServer.h"
 
 #include <signal.h>
@@ -46,6 +54,16 @@
         LogConfPtr conf(new LocalConf());
         LogConfMgr::instance().conf(conf);
     }
+#ifndef _WIN32
+	/* bug#24499 comment#4 */
+	struct rlimit nofile = {0,0};
+	if (getrlimit(RLIMIT_NOFILE, &nofile) == 0) {
+		if (nofile.rlim_cur > FD_SETSIZE) {
+			nofile.rlim_cur = FD_SETSIZE;
+			int ret = setrlimit(RLIMIT_NOFILE, &nofile);
+		}
+	}
+#endif
 }
 
 LogServer::~LogServer()
