diff -ru logtool-2.2.2.orig/src/LogFile.cc logtool-2.2.2/src/LogFile.cc
--- logtool-2.2.2.orig/src/LogFile.cc	2009-02-27 14:15:40 +0100
+++ logtool-2.2.2/src/LogFile.cc	2009-02-27 14:18:41 +0100
@@ -16,6 +16,12 @@
 #endif /* ! _WIN32 */
 #include "_logtool.h"
 
+#ifdef __PARITY__
+# include <internal/pcrt.h>
+#else
+# define PCRT_CONV(x) x
+#endif
+
 namespace wamas {
     namespace platform {
         namespace logging {
@@ -47,7 +53,7 @@
 #ifdef _WIN32
 static FILE* fileopen(char const* name)
 {
-	HANDLE winFile = CreateFile(name
+	HANDLE winFile = CreateFile(PCRT_CONV(name)
 		, GENERIC_WRITE
 		, FILE_SHARE_DELETE | FILE_SHARE_READ | FILE_SHARE_WRITE
 		, NULL
