--- src/LogServer.cc.orig	2014-03-28 09:48:46.728770698 +0100
+++ src/LogServer.cc	2014-03-28 09:48:13.188771208 +0100
@@ -148,10 +148,10 @@
         long ldiff=(long) diffclock;
 
         time_t now=time(0);
-        if( ldiff >500 && ( now - outputtime ) >1 ) {
-        	outputtime=time(0);
-            std::ostringstream cycletime;
-        	cycletime<<" cycletime greater than 500 ms! it is: "<<ldiff<<" [ms]";
+        if( ldiff >1500 && ( now - outputtime ) >60 ) {
+		outputtime=time(0);
+		std::ostringstream cycletime;
+        	cycletime<<" cycletime greater than 1500 ms! it is: "<<ldiff<<" [ms]";
         	writeError(cycletime.str());
         }
     }
