Index: src/LogFile.cc
===================================================================
RCS file: /cvsroot/saloon/wamas_x/utilities/logging/cpp/logtool/src/LogFile.cc,v
retrieving revision 1.21
retrieving revision 1.23
diff -u -r1.21 -r1.23
--- src/LogFile.cc	14 May 2012 13:17:53 -0000	1.21
+++ src/LogFile.cc	2 Jul 2012 14:14:22 -0000	1.23
@@ -102,6 +102,7 @@
             return (false);
         }
         if ( mkdirp(path) < 0) {
+            free(path);
             return (false);
         }
         free(path);
@@ -110,6 +111,16 @@
     }          
     
     if ( fp_ == 0 ) {
+#ifndef _WIN32
+        // filesize exceeded 2GB?
+        if (errno == EOVERFLOW
+#ifdef EFBIG /* linux before kernel-2.6.24 */
+         || errno == EFBIG
+#endif
+        ) {
+            return role();
+        }
+#endif
         return (false);
     }
     
