diff -ru -x '*.P*o' logtool-2.2.2.orig/src/LocalDriver.cc logtool-2.2.2/src/LocalDriver.cc
--- logtool-2.2.2.orig/src/LocalDriver.cc	2009-09-23 16:03:11.128209934 +0200
+++ logtool-2.2.2/src/LocalDriver.cc	2009-09-23 16:05:29.931236856 +0200
@@ -5,6 +5,7 @@
 
 #include "LocalDriver.h"
 #include "LogFile.h"
+#include <cstdio>
 
 namespace wamas {
     namespace platform {
diff -ru -x '*.P*o' logtool-2.2.2.orig/src/LogDriver.cc logtool-2.2.2/src/LogDriver.cc
--- logtool-2.2.2.orig/src/LogDriver.cc	2009-09-23 16:03:11.128209934 +0200
+++ logtool-2.2.2/src/LogDriver.cc	2009-09-23 16:03:29.181294598 +0200
@@ -6,6 +6,7 @@
 #include "LogDriver.h"
 
 #include <sstream>
+#include <cstdio>
 
 #include <wamas/platform/logging/levels.h>
 #include "_logtool.h"
diff -ru -x '*.P*o' logtool-2.2.2.orig/src/LogFile.cc logtool-2.2.2/src/LogFile.cc
--- logtool-2.2.2.orig/src/LogFile.cc	2009-09-23 16:03:11.128209934 +0200
+++ logtool-2.2.2/src/LogFile.cc	2009-09-23 16:05:55.344153740 +0200
@@ -16,6 +16,8 @@
 #endif /* ! _WIN32 */
 #include "_logtool.h"
 
+#include <climits>
+
 #ifdef __PARITY__
 # include <internal/pcrt.h>
 #else
diff -ru -x '*.P*o' logtool-2.2.2.orig/src/LogToolError.h logtool-2.2.2/src/LogToolError.h
--- logtool-2.2.2.orig/src/LogToolError.h	2009-09-23 16:03:11.127360700 +0200
+++ logtool-2.2.2/src/LogToolError.h	2009-09-23 16:03:54.277201313 +0200
@@ -6,6 +6,8 @@
 #ifndef _WAMAS_PLATFORM_LOGGING_LOGTOOLERROR_H
 #define _WAMAS_PLATFORM_LOGGING_LOGTOOLERROR_H
 
+#include <cstring>
+
 namespace wamas {
     namespace platform {
         namespace logging {
diff -ru -x '*.P*o' logtool-2.2.2.orig/src/RemoteDriver.cc logtool-2.2.2/src/RemoteDriver.cc
--- logtool-2.2.2.orig/src/RemoteDriver.cc	2009-09-23 16:03:11.130356783 +0200
+++ logtool-2.2.2/src/RemoteDriver.cc	2009-09-23 16:05:08.739199859 +0200
@@ -4,6 +4,7 @@
  */
 
 #include "RemoteDriver.h"
+#include <cstdio>
 
 namespace wamas {
     namespace platform {
diff -ru -x '*.P*o' logtool-2.2.2.orig/test/test_forkedClient.cc logtool-2.2.2/test/test_forkedClient.cc
--- logtool-2.2.2.orig/test/test_forkedClient.cc	2009-09-23 16:03:11.134151625 +0200
+++ logtool-2.2.2/test/test_forkedClient.cc	2009-09-23 16:06:39.824204194 +0200
@@ -11,6 +11,9 @@
 
 #include <logtool.h>
 
+#include <cstdio>
+#include <cstring>
+
 #define MAX_ORIGIN      10
 #define MAX_FACILITY    10
 #define MAX_LEVEL       6
