diff -ru logtool-2.2.2/src/LogFile.cc logtool-HEAD/src/LogFile.cc
--- logtool-2.2.2/src/LogFile.cc	2012-07-09 09:42:07.342686152 +0200
+++ logtool-HEAD/src/LogFile.cc	2012-07-02 16:10:31.249623947 +0200
@@ -270,7 +281,7 @@
 #if defined(_WIN32)
 FILETIME LogFile::getHandleTime()
 {
-        FILETIME time;
+        FILETIME time = {0,0};
         HANDLE hFile = (HANDLE)_get_osfhandle(fileno(fp_));
 
         if(!hFile || hFile == INVALID_HANDLE_VALUE)
@@ -289,7 +300,7 @@
 FILETIME LogFile::getFilePhysTime()
 {
         WIN32_FILE_ATTRIBUTE_DATA data;
-        FILETIME time;
+        FILETIME time = {0,0};
 
         if(!GetFileAttributesEx(fullName_.c_str(), GetFileExInfoStandard, &data))
         {
@@ -349,14 +360,20 @@
 	/*
 	 * del files > MAX_ARCHIVE_FILES
 	 */
-	for( i = MAX_ARCHIVE_FILES;i >= maxFiles_; --i) {
+	i = MAX_ARCHIVE_FILES+1;
+	do {
+		/*
+		 * Need to unlink some files even if maxFiles_ is
+		 * either larger than MAX_ARCHIVE_FILES, or zero.
+		 */
 		sprintf(archive_file,"%s.%d",fullName_.c_str(), i-1);
 #if !defined(_WIN32)
 			unlink(archive_file);
 #else /* _WIN32 */
 			_unlink(archive_file);
 #endif /* _WIN32 */
-	}
+		--i;
+	} while(i > maxFiles_);
 	
 	/*
 	 * rename file.(x) -> file.(x+1)
