diff -ruN logtool-2.2.2.orig/src/LogConnection.cc logtool-2.2.2/src/LogConnection.cc
--- logtool-2.2.2.orig/src/LogConnection.cc	2009-01-22 11:49:09.000000000 +0100
+++ logtool-2.2.2/src/LogConnection.cc	2009-02-10 14:12:41.000000000 +0100
@@ -35,7 +35,6 @@
 	, addr_()
 	, connected_(false)
 	, data_()
-	, root_()
 	, logRoot_()
 {
     char rootDir[PATH_MAX+1];
@@ -43,8 +42,6 @@
 
     memset(&addr_, 0, sizeof(Addr));
 
-    /* get ROOTDIR/LOGROOT */
-    root_ =  getcwd(rootDir, PATH_MAX+1);
     tmp = getenv(LT_ENV_ROOTDIR);
     if ( tmp  != 0 ) {
         logRoot_ = tmp;
@@ -241,14 +238,26 @@
 
 bool LogConnection::doConnect(size_t addrLen)
 {
+    char rootDir[PATH_MAX+1];
     
+    /* get ROOTDIR/LOGROOT */
+    std::string currentwd = getcwd(rootDir, PATH_MAX+1);
+
     if ( !logRoot_.empty() ) {
         if (chdir(logRoot_.c_str()) < 0) {
             writeError("(ignored) chdir to LOGROOT");
         }
     }
 
-    if (SockConnect(sock_, &addr_.A, addrLen) == SOCKET_ERROR) {
+	int rv = SockConnect(sock_, &addr_.A, addrLen);
+
+    if ( !logRoot_.empty() && !currentwd.empty() ) {
+        if (chdir(currentwd.c_str()) < 0) {
+            writeError("(ignored) chdir to cwd");
+        }
+    }
+
+	if (rv == SOCKET_ERROR) {
 		int     error, trials = 0;
         char    string[256];
 
@@ -291,12 +300,6 @@
         fcntl(sock_, F_SETFD, 1);
 #endif /* ! _WIN32 */
 
-    if ( !logRoot_.empty()  && !root_.empty() ) {
-        if (chdir(root_.c_str()) < 0) {
-            writeError("(ignored) chdir to cwd");
-        }
-    }
-
     return (true);    
 }
 
diff -ruN logtool-2.2.2.orig/src/LogConnection.h logtool-2.2.2/src/LogConnection.h
--- logtool-2.2.2.orig/src/LogConnection.h	2008-01-30 11:48:08.000000000 +0100
+++ logtool-2.2.2/src/LogConnection.h	2009-02-10 14:12:25.000000000 +0100
@@ -68,7 +68,6 @@
     bool connected_;
     std::ostringstream data_;
     
-    std::string root_;
     std::string logRoot_;
 };
         
