--- logtool-2.2.2/src/LogClientLogger.cc.orig	2011-09-15 11:05:13.400856399 +0200
+++ logtool-2.2.2/src/LogClientLogger.cc	2011-09-15 11:06:09.410847049 +0200
@@ -225,11 +225,11 @@
     
     version = prot_->version();
     if ( version.compare(V10) == 0 ) {
-	  do {
         Protocol_V10* protV10 = dynamic_cast<Protocol_V10*>(prot_.get());
         std::stringstream data;
         std::string request;
         bool finished = false;
+		unsigned int tries = 0;
 
         protV10->type(LOGCONF_REQUEST);
         request = protV10->generate();
@@ -240,7 +240,8 @@
             return false;
         }
         
-        while ( !finished ) {
+        while ( confUpdateRequested_ || !finished ) {
+			finished = false;
             if ( conn_.receive() ) {
                 if ( parseData() ) {
                     finished = true;
@@ -253,9 +254,16 @@
                 timeout.tv_usec = 10000; /* 10ms */
 
                 SockSelect(0, NULL, NULL, NULL, &timeout);
+				if (++tries > 300) {
+					/* give approx. 3 seconds */
+					conn_.close();
+					// use LocalConf now in local mode
+					LogConfPtr conf(new LocalConf());
+					LogConfMgr::instance().conf(conf);
+					return (false);
+				}
             }
         }
-	  } while (confUpdateRequested_);
 	} else {
         return (false);
     }
